internal/analysis/analysis.go

```diff
@@ -0,0 +1,58 @@
+package analysis
+
+import "fmt"
+
+// Schema defines a type for a schema, a mapping of key names to their types.
+// Types supported are: "number", and "bool".
+type Schema map[string]string
+
+// Data provides a type for data, a map of key names to their values.
+type Data map[string]interface{}
+
+// Results provides a type for analysis results, a map of key names to the
+// results (ex. averages).
+type Results map[string]float64
+
+func btoi(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+// Average averages all "number" and "bool" fields. True is considered 1, and false
+// is considered zero.
+func Average(schema Schema, datas ...Data) (Results, error) {
```

Comment:
Nope. Just nope. I refuse to use `datas`. We either use `data` for both plural and singular (I think golang for loops are a separate scope so we can just shadow inside the loop) or use `datum` for the singular?

Reply:
`datum` is such a fun word. Let's use that :)